### PR TITLE
env-check.sh: rewrite complicated and duplicated group membership

### PR DESCRIPTION
**Pull request by marc-hb**
Simplify and remove duplication, use an array and for loop instead. 450
fewer characters.
Stop parsing files directly, switch to getent system interface instead.
Stop recommending editing file directly, recommend usermod system
interface instead.

Fix the last shellcheck warnings in this file:

In env-check.sh line 107:
```
[[ "$user" ]] && [[ ! $(awk -F ':' '/^adm:/ {print $NF;}' /etc/group|grep "$user") ]] && \
                      ^-- SC2143: Use grep -q instead of comparing output with [ -n .. ].
```
(3 times because of code duplication)

Signed-off-by: Marc Herbert <marc.herbert@intel.com>

### CODE REVIEW

**Comment by marc-hb** [comment]
The last shellcheck warnings in this file are fixed in (unrelated) PR #452

File: env-check.sh [hunk 1]
```diff
@@ -94,32 +94,32 @@ echo -ne "Checking for case folder:\t\t"
...
-\tMissing permission to run command as sudo\n
-\t\tPlease use the following command to add current user to the sudo group:\n
-\t\e[31msed -i '/^sudo:/s:$:,$user:g' /etc/group\e[0m"
+declare -A grp_error
+grp_error=(
+[adm]='\tMissing permission to access /var/log/kern.log\n'
+[sudo]='\tMissing permission to run command as sudo\n'
+[audio]='\tMissing audio group membership to access /dev/snd/* devices\n'
+)
+for grp in "${!grp_error[@]}"; do
```
Comment (xiulipan) [inline env-check.sh:113]: why not wrap this as a function.

```
check_group adm 'Missing permission to access /var/log/kern.log'
check_group sudo 'Missing permission to run command as sudo'
check_group audio 'Missing audio group membership to access /dev/snd/* devices' ```

**Review by xiulipan — CHANGES_REQUESTED** [review]
Nice idea but I think it would be better with some function wrapper.

File: env-check.sh (hunk 1 above)
Comment (marc-hb) [reply env-check.sh]: Good idea, thanks! Done.

**Review by xiulipan — APPROVED** [review]
LGTM

**Pull request merged**